src/Substrait.Lineage/LineageExtractorRelVisitor.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Substrait.Core;
+using Substrait.Relation;
+using ProtoRel = Substrait.Protobuf.Rel;
+
+namespace Substrait.Lineage
```

Comment:
Hmm...I'm maybe a little confused here.  What is the intent of this project?  Isn't lineage an implicit property of `Rel`?  I would expect this project would be extending `Rel` somehow to add lineage information to it (or extracting lineage information into some kind of project-specific classes).

It seems a bit odd that "grabbing lineage for one relation" is a `Substrait.Core` task but "grabbing lineage for an entire relation tree" is a `Substrait.Lineage` task.  I would think either both are `Substrait.Lineage` or both are `Substrait.Core` (with a minor preference for the latter)